common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -632,6 +635,25 @@ def get_problem_html(self, encapsulate=True):
             'weight': self.weight,
         }
 
+        # Handle demand hints
+        demand_hints = self.lcp.tree.xpath("//problem/demandhint/hint")
+        show_hint_button = len(demand_hints) > 0
+        demand_hint = None  # if set to something, will show up in UI
+        if hint_index is None:
+            hint_index = 0  # bootstrap case, the client will start at 0
+        elif len(demand_hints) > 0:
+            _ = self.runtime.service(self, "i18n").ugettext  # pylint: disable=redefined-outer-name
+            hint_element = demand_hints[hint_index]
+            if len(demand_hints) == 1:
+                prefix = _('Hint: ')
+            else:
+                # Translators: e.g. "Hint 1 of 3" meaning we are showing the first of three hints.
+                # Do not translate hint_num and hints_count
+                prefix = _('Hint ({hint_num} of {hints_count}): ').format(hint_num=hint_index + 1,
```

Comment:
[See also Python docs](https://docs.python.org/2/library/gettext.html#gettext.ngettext)

You do ngettext (in English) as follows: `ungettext(singular, plural, number_on_which_to_switch)`. So for example you might do
```python
numf = get_number_of_files()
# This line does the ngettext call but does NOT do the format of the 'num_files' var
msg = ungettext("There is {num_files} file", "There are {num_files} files", numf)
# Now once the ngettext call has returned with the proper translation, do the variable substitution with .format
msg = msg.format(num_files=numf)
```